### PR TITLE
[REF] change single class use case

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### What does this implement/fix? Explain your changes.

We have an internal mechanism in base classifier for the edge case when all the y values seen in fit are the same. The predict and predict_proba then just return that value or prob = 1 of that value.

this was done by a call to check_is_scitype, which involves multiple scans of the data. This simplifies it by using len, except for the edge case of using multi index DataFrame. 

Also removed a pointless duplication in tests and moved the overloaded operator to the bottom of the file and changed series_length to n_timepoints.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
> Changes to `setup.cfg` are included in the PR 🙂

*sigh*, fixed :)

File: aeon/classification/base.py [hunk 1]
```diff
@@ -85,54 +85,20 @@ def __init__(self):
         super(BaseClassifier, self).__init__()
         _check_estimator_deps(self)
 
-    def __rmul__(self, other):
-        """Magic * method, return concatenated ClassifierPipeline, transformers on left.
-
```
Comment (MatthewMiddlehurst) [inline aeon/classification/base.py:90]: I think it is neater to keep the `__X__` type methods at the top, even if we are not the biggest fans of the contents 🙂

File: aeon/classification/base.py [hunk 2]
```diff
@@ -85,54 +85,20 @@ def __init__(self):
...
         X : 3D np.array (any number of channels, equal length series)
-                of shape [n_instances, n_channels, series_length]
+                of shape (n_instances, n_channels, n_timepoints)
             or 2D np.array (univariate, equal length series)
-                of shape [n_instances, series_length]
-        y : 1D np.array of int, of shape [n_instances] - class labels for fitting
+                of shape (n_instances, n_timepoints)
+            or list of numpy arrays (any number of channels, unequal length series)
+                of shape [n_instances], 2D np.array n_channels, n_timepoints_i), where
+                n_timepoints_i is length of series i
```
Comment (MatthewMiddlehurst) [inline aeon/classification/base.py:99]: This looks like it's missing a ( in the middle

Same for other cases where this comment appears

File: extension_templates/classification.py [hunk 3]
```diff
@@ -97,10 +97,13 @@ class MyTimeSeriesClassifier(BaseClassifier):
         "python_version": None,  # PEP 440 python version specifier to limit versions
     }
 
-    # todo: add any hyper-parameters and components to constructor
+    # todo: add any parameters to constructor
     def __init__(self, param_a, param_b="default", param_c=None):
         # estimators should precede parameters
         #  if estimators have default values, set None and initalize below
+        # Note that parameters passed and set in constructor should not be changed
+        # in order to comply withe scikit learn strucutre. Intead, copy into local
```
Comment (MatthewMiddlehurst) [inline extension_templates/classification.py:105]: Typo

File: aeon/classification/base.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/classification/base.py:90]: I looked, it's not a convention that I could see, I prefer it out the way

Comment (TonyBagnall) [reply aeon/classification/base.py:90]: I mean its not a python convention afaik, aeon conventions are inherited from the bad place. But if its a blocker, I'll move it back

File: aeon/classification/base.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/classification/base.py:99]: good spot

File: extension_templates/classification.py (hunk 3 above)
Comment (MatthewMiddlehurst) [reply extension_templates/classification.py]: "in order to comply withe scikit learn"

**Review by MatthewMiddlehurst — APPROVED** [review]
